core/shared/src/main/scala/org/http4s/Method.scala

```diff
@@ -75,6 +77,11 @@ object Method {
   private def safe(name: String) =
     new Method(name, isSafe = true, isIdempotent = true)
 
+  private val NoPayloadMethods: Set[Method] =
+    Set(Method.GET, Method.DELETE, Method.CONNECT, Method.TRACE)
```

Comment:
What about `HEAD` and `OPTIONS`?

Reply:
There is a table of methods in the wiki page, describing the payload body as either "yes", "optional", or "no".
https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Request_methods

Reply:
That is to say, perhaps the name of the method should be `payloadRequired` ?